script/DeployVault.sol

```diff
@@ -0,0 +1,210 @@
+import "src/delayed-withdrawal-manager/DelayedWithdrawalManager.sol";
+import "src/strategies/layers/creation-validation/external/TOSManager.sol";
+import "src/strategies/layers/creation-validation/external/GlobalValidationManagersRegistry.sol";
+import "src/strategies/layers/creation-validation/external/SignatureBasedWhitelistManager.sol";
+import "@forta/firewall/ExternalFirewall.sol";
+import "@forta/firewall/FirewallAccess.sol";
+import "@forta/firewall/SecurityValidator.sol";
+import "@forta/firewall/FirewallRouter.sol";
+
+import "@forta/firewall/interfaces/ICheckpointHook.sol";
+import "@forta/firewall/interfaces/Checkpoint.sol";
+
+import "src/strategies/instances/aave-v3/AaveV3Strategy.sol";
+import "src/strategies/instances/aave-v3/AaveV3StrategyFactory.sol";
+
+import "src/strategies/instances/erc4626/ERC4626Strategy.sol";
+import "src/strategies/instances/erc4626/ERC4626StrategyFactory.sol";
+
+import "src/strategies/instances/lido/LidoSTETHStrategy.sol";
+import "src/strategies/instances/lido/LidoSTETHStrategyFactory.sol";
+
+import "./BaseDeploy.sol";
+
+contract DeployVault is BaseDeploy {
+  function run() external {
+    address signer = vm.envAddress("SIGNER");
+    vm.startBroadcast(deployerPrivateKey);
+    address strategyRegistry =
+      deployContract("V1_STRATEGY_REGISTRY", abi.encodePacked(type(EarnStrategyRegistry).creationCode));
+    console2.log("Strategy registry:", strategyRegistry);
+    address nftDescriptor = deployContract(
+      "V1_NFT_DESCRIPTOR",
+      abi.encodePacked(
+        type(EarnNFTDescriptor).creationCode, abi.encode("http://api.balmy.xyz/v1/earn/metadata/", admin)
+      )
+    );
+    console2.log("NFT descriptor:", nftDescriptor);
+    address[] memory initialAdmins = new address[](2);
+    initialAdmins[0] = admin;
+    initialAdmins[1] = deployer;
+
+    // FORTA
+    bytes32 attesterControllerId = bytes32("3");
+    ISecurityValidator validator = ISecurityValidator(0xc9b1AeD0895Dd647A82e35Cafff421B6CcFe690C);
+
+    address firewallAccess =
+      deployContract("V1_FACCESS", abi.encodePacked(type(FirewallAccess).creationCode, abi.encode(deployer)));
+    address externalFirewall = deployContract(
+      "V1_FEXTERNAL",
+      abi.encodePacked(
+        type(ExternalFirewall).creationCode,
+        abi.encode(validator, ICheckpointHook(address(0)), attesterControllerId, firewallAccess)
+      )
+    );
+    address firewallRouter = deployContract(
+      "V1_FROUTER",
+      abi.encodePacked(
+        type(FirewallRouter).creationCode,
+        abi.encode(ExternalFirewall(externalFirewall), FirewallAccess(firewallAccess))
+      )
+    );
+    address vault = deployContract(
+      "V1_VAULT",
+      abi.encodePacked(
+        type(FirewalledEarnVault).creationCode,
+        abi.encode(IEarnStrategyRegistry(strategyRegistry), admin, initialAdmins, nftDescriptor, firewallRouter)
+      )
+    );
+    address permit2Adapter = 0xED306e38BB930ec9646FF3D917B2e513a97530b1;
```

Comment:
The permit2 adapter can vary from chain to chain. We should handle this differently